### ISSUE TITLE
Display the current test condition

### ISSUE THREAD

**Issue by haramoz** [maxpumperla/hyperas#191]
Hi ,

I need to ask you how to display the current test-condition for the optimization. For I am constantly bugged by cluster memory issues. My jobs run out of memory some times(working with Densenet ...) after 2 days of execution. And then all this two days worth results are just worth-less because the configs for the tests that are already done are unknown to me. 
Once the best run, best model is produced I extract the details of all the run like this.
        
       ```for i in trials.trials:
        vals = i.get('misc').get('vals')        
        results = i.get('result').get('loss')
        print(vals,results)```
I am trying to counter that with as follows:
 ``` 
    try:
         best_run, best_model = optim.minimize(...)
    except:
        #if exception stops the grid search
        print(trials.trials)```

This is not performing that reliably, so how do I display the current test condition before/after the test starts? if this is not clear so far, I would like to get some thing like, 
current test condition: `{'depth': [0], 'growth_rate': [1], 'nb_dense_block': [0]}` ...

Is there a way for this? Please let me know. If this already exists then cool, if not, then perhaps this is more like a feature request. Thank you very much :)

**Comment by dendrondal** [comment]
I've been having the same issue with one of my models. I do three things to address this, one of which is the try/except clause like you have above. However, I still have issues with this. Currently, I have something along these lines:
```
from contextlib import redirect_stdout
import logging
from keras import backend as K

def model(X_train, X_test, y_train, y_test):
     #model definition/compilation goes here
     logging.basicConfig(filename='models.log')
     #I save hyperparameters as I go, especially helpful if it crashes
     with open('modelsummary.txt', 'w') as f:
         with redirect_stdout(f):
             model.summary()
    try:
        model.fit(**kwargs)
    except error as e:
         logging.warning(e.__doc__)
         #This just dumps the stack trace, isn't overly useful.
         #I thought the statements below would keep it from hanging, but no luck
         K.clear_session()
         pass 

   return {'loss': -acc, 'status': STATUS_OK}
    
```
This results in the following error:

> WARNING:root:Some resource has been exhausted.
> 
>   For example, this error might be raised if a per-user quota is
>   exhausted, or perhaps the entire file system is out of space.
> 
>   @@__init__
>   
> INFO:hyperopt.fmin:job exception: Attempting to use uninitialized value 
>**TF stacktrace goes here**
>FailedPreconditionError (see above for traceback): Attempting to use uninitialized value

I'd like to work on an improvement for this, but I'm not sure how to go about it. I'll test some changes to optim.minimize and see if I can submit a PR which allows for the optimization to skip models that are too big to fit in memory.

**Comment by haramoz** [comment]
Alright, i see whats going on there from your side. From my side I am still confused where should I place the K.clear_session() to make it function. I tried putting it at the start of the create_model function, also at the end. Still my code crashes after some evaluations. 
For getting the current test conditions though I found a simple work around, I am just printing the value of the variables (i.e those have choice and uniform values) **inside the create_model function.**  That way I can retrieve the result for the current test conditions. So I am closing this issue but need to open another for bad_alloc error.
